lessons/5.mdx

```diff
@@ -0,0 +1,1362 @@
+
+npx create-next-app frontend --use-npm
+```
+
+Once the setup is complete, we open the `frontend` directory and get cracking.
+
+```bash
+cd frontend
+```
+<br/>
+
+---
+
+## The Frontend
+
+For users to easily interact with our smart contract and mint the NFTs we must
+provide an interface. A simple process that is easily accessible.
+
+Let's start by creating a way to handle wallet connections.
+
+### The Main Application
+
+We will use [RainbowKit](https://www.rainbowkit.com/docs/introduction) for handling
+the wallet connections. It provides us the ability to handle wallet connections, 
+support for numerous wallets and a customisable `ConnectButton` out-of-the-box among 
+other features. It relies on the [WAGMI](https://wagmi.sh/) and [ethers](https://docs.ethers.io/v5/)
+libraries for interoperability between various chains and wallet providers and 
+thus needs some configurations.
+
+But first we need to install the necessary dependencies.
+
+```bash
+yarn add @rainbow-me/rainbowkit wagmi ethers
+```
+
+To get this setup we need to start working on the `_app.js` file in the `/frontend/pages`
+directory. The file should look something like this from our initial setup.
+
+```jsx
+import '../styles/globals.css'
+
+function MyApp({ Component, pageProps }) {
+  return <Component {...pageProps} />
+}
+
+export default MyApp
+```
+
+We need to make some additional imports to create our configurations.
+
+```jsx
+// previous imports...
+import '@rainbow-me/rainbowkit/styles.css';
+import {
+  getDefaultWallets,
+  RainbowKitProvider,
+} from '@rainbow-me/rainbowkit';
+import {
+  chain,
+  configureChains,
+  createClient,
+  WagmiConfig,
+} from 'wagmi';
+import { alchemyProvider } from 'wagmi/providers/alchemy';
+import { publicProvider } from 'wagmi/providers/public';
+import { MainContext } from '../context';
+import { useEffect, useState } from 'react';
+
+// function MyApp code...
+```
+
+- `import '@rainbow-me/rainbowkit/styles.css';` gives us some basic styling from
+Rainbowkit.
+- `import { getDefaultWallets, RainbowKitProvider, } from '@rainbow-me/rainbowkit';` returns
+the default wallet provider option and a RainbowKit Provider to "wrap" around our app
+so that we can use its features throughout the app's pages and components.
+- `import { chain, configureChains, createClient, WagmiConfig, } from 'wagmi';` provides
+a way to configure the WagmiConfig wrapper for our app based on the chains and providers
+of our choice along with some other customisations.
+- `import { alchemyProvider } from 'wagmi/providers/alchemy';` and 
+`import { publicProvider } from 'wagmi/providers/public';` give us the provider configs
+for the providers we will be using. In our app we are using an `alchemyProvider` along
+with a fall back `publicProvider`. 
+- `useEffect` and `useState` are react hooks that help us perform side effects and
+capture the state, respectively.
+
+After importing we create the configurations we need and create a new instance of 
```

Comment:
After importing, we create the configurations we need, and create a new instance of of `wagmiClient` using them.